pkg/apis/application/v1alpha1/types.go

```diff
@@ -621,6 +623,23 @@ func (c *ApplicationSourcePlugin) RemoveEnvEntry(key string) error {
 	return fmt.Errorf("unable to find env variable with key %q for plugin %q", key, c.Name)
 }
 
+// CopyFromSpec defines a copy declaration from another source into this one
+type CopyFromSpec struct {
+	// SourcePath is the path from the Ref'd repository, the source of the copy operation.
+	// It's syntax is matching helmValue's, $ref/path
+	SourcePath string `json:"sourcePath" protobuf:"bytes,2,name=sourcePath"`
+	// DestinationPath is the destination directory at the current repository where the object will be copied to.
+	DestinationPath string `json:"destinationPath" protobuf:"bytes,3,name=destinationPath"`
+	// Hardfail mode makes the complete operation error out if a single symlink is out-of-bound within the subtree
+	FailOnOutOfBoundsSymlink bool `json:"failOnOutOfBoundsSymlink,omitempty" protobuf:"bytes,4,name=failOnOutOfBoundsSymlink"`
```

Comment:
Yes, This used to have 3 fields, the first one a `ref`, which @crenshaw-dev suggested to be removed. And protobuf says if a field is to be removed, its number should be skipped.

Since this never got into upstream yet, I don't think renumbering the fields would break anything. But that's the reason why it doesn't start with 1.

Reply:
Fixed in `bbd96c6b`.